All_home_works/hw2/task4.py

```diff
@@ -0,0 +1,36 @@
+"""
+Write a function that accepts another function as an argument. Then it
+should return such a function, so the every call to initial one
+should be cached.
+
+
+def func(a, b):
+    return (a ** b) ** 2
+
+
+cache_func = cache(func)
+
+some = 100, 200
+
+val_1 = cache_func(*some)
+val_2 = cache_func(*some)
+
+assert val_1 is val_2
+
+"""
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    memo = {}
+
+    def wrapper(*args, **kwargs):
+        call = (args, tuple(kwargs))
```

Comment:
How about mutable arguments, why do you droped their support?

If we use only immutable arguments what could be wrong?  (give me 2 examples of different `*args`, `**kwargs` which will be different by will produce that same `call`) (Find it, add tests for it, fix it)

Reply:
I made mistake, i wanted: call = (args, tuple(kwargs.items())).
In previous code kwargs not worked at all.

For this code: call = (args, tuple(kwargs.items())).
foo(1, 2.0, 0)
foo(True, 2, False)

And except for such easy examples, unfortunately, I can't suggest more examples.